### DISCUSSION TITLE
storymode & arcadepath limitations

### DISCUSSION THREAD

**Discussion by ZyonZZ** [ikemen-engine/Ikemen-GO#1108] [Feature Request]
### Is your feature request related to a problem? Please describe.

Although there aren't many tutorials on the internet on,. how to use this function in an advanced way... with the tutorial added to ikemen I've been able to try both modes and they're good...

### Describe the solution you'd like

I would like to suggest a new feature for the "storymodeADV"..
I would like to be able to have the advantages of storymode such as "storyboards" that it allows as well as the selectcharacter that it has... how would this be...

Let's imagine that... we make a dragonball IKEMEN... then I prepare the ARC in the menu... with the following arcs

1-Sayan Saga
2-namek saga
3-Android saga
4-Boo Saga

So in said menu these arches would appear... then I select... "Namek Saga" and immediately afterwards the Character Select is displayed and the teammode would start as
I select Raditz.....storyboard...namekintro.def.......and fight order1, 2, 3, 4.... I advance through the saga until I reach combat5 Frieza a boss end storyboard..namekending.def
end
((allowing as optional if you want it to use the data from the Select.def))

Now if I want the other Saga of the androids... I select their saga from the menu... character select... Raditz....storyboard...androidsintro.def.... and the order would be 6, 7, 8, and the boss order=9 Boss Cell,
((These order numbers 6,7,8,9 are exclusively for saga androids))
 ......storyboard-----endingandroid.def
choosing the character that I want, and if I want them to be double battles or turns...
and it is that either ArcadePath or Storyboard are currently linked to a single character... and I was looking for different sagas, with and choose any character without limitation..

It would be perfect to be able to implement this online coop but that is already dreaming xD

I hope to make myself understood in the points, I talked about it with people who program at ikemen and they told me that the storyboard thing is still in beta, but I see it as only little details are missing... and something very good can be done

It really is looking better ikemen than mugen, greetings : D

**Comment by Orochikyocr** [comment]
It is not wise for devs to be covering every single need people came out with. 
Also
1- About the sagas, you can already do this with the story path feature.
2- There are already arcade path tutorials. Here is one. https://costaricaonfoot.blog/2022/11/29/story-path-for-ikemen-go/

I highly encourage you to create your arcade paths and share them with other users. Many people ask for help to create stuff but never share back, which is sad.

**Reply by ZyonZZ** [reply]
I already knew that blog (costaricaonfoot.blog) but the point is that the storymode is linked to the character... that's the whole problem... if you could change that variable at will or leave it optional for example...


-------------------------------------------------- -------------------------------------------------
launchFight

p1char = {list} (table)
List of characters (names/paths enclosed in quotation, separated by commas) to assign to p1 team side.

     default arcade path: characters chosen in select screen
     default story mode: previously assigned chars

p2teammode = "mode" (string)
Assigns p2 team mode to single, simul, tag or turns.

     default arcade path: team mode chosen during select screen
     default story mode: single
-------------------------------------------------- -------------------------------------


It's basically removing these limitations in the Storymode and allowing the character screen to be able to be changed. There really isn't much to modify.

As I tell you, this is to give versatility and not link everything to the chars but rather to a SAGA or History... with the character you choose... say in fighting games that already exemplify such as DBZ or Naruto games where you play the ARCs

**Comment by Orochikyocr** [comment]
Story mode. It is in the wiki 
```1- About the sagas, you can already do this with the story path feature.```
Story is not sticked to any char, you can force the char the player has to use like Smash story mode.

**Reply by ZyonZZ** [reply]
where? as? when? link of that info... Plz

**Comment by Orochikyocr** [comment]
https://github.com/ikemen-engine/Ikemen-GO/wiki/Miscellaneous-info#arcade-paths--story-mode
Story Mode Arcs (selectable from the main menu as part of Story Mode, e.g. cinematic stories where the author decides exactly who the player fights in each match, and against who, with multiple storyboards displayed along the way)
![image](https://user-images.githubusercontent.com/42851314/231004681-f21a741b-1f46-4dbb-b3e0-d2b05f174564.png)
Ad your story modes in the select.def. Look for it in the default select.def that comes with any clean Ikemen Go build.

**Reply by ZyonZZ** [reply]
If you use the code you will realize that they are linked to the character... as I already exemplified here...
What you are saying is the theory.

the practice is this...
launchFight

p1char = {list} (table)
List of characters (names/paths enclosed in quotation, separated by commas) to assign to p1 team side.

  default arcade path: characters chosen in select screen
  default story mode: previously assigned chars

p2teammode = "mode" (string)
Assigns p2 team mode to single, simul, tag or turns.

  default arcade path: team mode chosen during select screen
  default story mode: single

and here they are clearly tied to the character... and there is no way to create sagas.
This method is thought of as a SF game where each character has its own story.

**Comment by Orochikyocr** [comment]
What code are you talking about? Story mode HAS NOT "character select" screen. With story mode, you force the player to use a specific char.
You're mistaken and it only shows you have not tried "story mode" at all. 

Story mode
1- The player cannot choose characters because there is no "character select" screen.
2- All fights are defined in the LUA file, the default method of selecting the CPU opponent does not work here. If your story lua file is blank or unfinished, the engine will not assign random opponents to the player. You have full control of what happens in this mode.
3- You can have several story.lua files and set them in the select.def, there you can have "SAGAS", different stories inside your game.

The engine is ready to create full story mode like any commercial Dragonball game out there.

 You should try first before saying is not possible. 

Story mode is actually WAY easier than arcade path.

if matchno() == 1 then
if not launchFight{
        p1char = "Goku",
        p2char = "Radditz",
        p2numchars = 1,
        p2teammode = "single"
    }
 }
if matchno() == 2 then
if not launchFight{
        p1char = ("Piccolo","Gohan","Krillin"),
        p1numchars = 3,
        p1teammode = "tag",
        p2char = "Nappa",
        p2numchars = 1,
        p2teammode = "single"
    }   
 } 
 if matchno() == 3 then
if not launchFight{
        p1char = "Goku",
        p1numchars = 1,
        p1teammode = "single",
        p2char = "Nappa",
        p2numchars = 1,
        p2teammode = "single"
    }   
 } 
  if matchno() == 4 then
if not launchFight{
        p1char = "Goku",
        p1numchars = 1,
        p1teammode = "single",
        p2char = "Vegeta",
        p2numchars = 1,
        p2teammode = "single"
    }   
 } 

This is an untested example, but as I said you can use the arcade path I created and modify it to create yours.

**Reply by ZyonZZ** [reply]
Your answer is a clear example that you have not understood the meaning of the post.

I apologize.. maybe I have not been specific enough..

""1- The player cannot choose characters because there is no "character select" screen.""

this is exactly the problem..
I invite you to reread the first post and then come back here...



because the point of all this is that it is too predetermined and mandatory for specific combats..
therefore... it's like creating 5 sagas for raditz (androids saga, buu saga, cell saga, super saga)
5 sagas for goku
5 sagas for pikolo
5 sagas for frezzer
5 sagas for vegeta
5 sagas for ETC

5 sagas for each character that you put because the list would become immensely large... and nothing overly flashy to choose from...

for that very reason... remove the limitation of the Storymode to a character.
for when choosing the SAGA... to be able to choose the character to taste
